pkg/reconcilermanager/controllers/gitsync_env.go

```diff
@@ -163,11 +167,18 @@ func gitSyncEnvs(_ context.Context, opts options) []corev1.EnvVar {
 		Name:  GitSyncRepo,
 		Value: opts.repo,
 	})
-	// disable known_hosts checking because it provides no benefit for our use case.
-	result = append(result, corev1.EnvVar{
-		Name:  gitSyncKnownHosts,
-		Value: "false",
-	})
+
+	if opts.knownHost {
+		result = append(result, corev1.EnvVar{
+			Name:  GitSyncKnownHosts,
+			Value: "true",
+		})
+	} else {
```

Comment:
Perhaps add a comment here for future readers:
```
// This must default to false for backwards compatibility
```

Reply:
added